mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -154,6 +154,98 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 							Computed:    true,
 							Description: `When set to true, the action specified above is not enforced. Stackdriver logs for requests that trigger a preview action are annotated as such.`,
 						},
+
+					<% unless version == 'ga' -%>
+						"rate_limit_options": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: `Rate limit threshold for this security policy. Must be specified if the action is "rate_based_ban" or "throttle". Cannot be specified for any other actions.`,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"rate_limit_threshold": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: `Threshold at which to begin ratelimiting.`,
+										MaxItems:    1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"count": {
```

Comment:
Are either / both / one of these fields required? A block where both are unset maps to a weird case in Terraform, sometimes

Reply:
The GCP docs (see https://cloud.google.com/compute/docs/reference/rest/beta/securityPolicies/addRule ) don't explicitly say, so I'd assume both are optional.

However, to test it, I created new rules via GCP CLI with `gcloud beta compute security-policies rules create ...`, and provide `rate-limit-threshold-count` and `rate-limit-threshold-interval-sec`. If I don't explicitly provide values for both fields, I get the following error: `Rate limit threshold, conform action, and exceed action must be specified for rate limit options`.

To be safe, I'm going to update this to require all four of these values.

Reply:
Does that mean both subfields here should become required (they're currently optional) as well as the ones under `ban_threshold` or have I misinterpreted your statement?

Reply:
You're correct, sorry, I missed that. Fixed